docs/platform-engineers/addon/intro.md

```diff
@@ -100,175 +107,493 @@ metadata:
+If you want the end user to determine some fields of the application by setting parameters when enabling the addon, you need write a `template.cue` file to do so. You can define the `template.cue` file like this to allow end-user specify the name of namespace in the above example:
 
-#### Examples
+```cue
+package main
+
+output: {
+	apiVersion: "core.oam.dev/v1beta1"
+	kind:       "Application"
+	metadata: {
+		name:      "example"
+		namespace: "vela-system"
+	}
+	spec: {
+		components: [
+			{
+				type: "k8s-objects"
+				name: "example-namespace"
+				properties: objects: [{
+					apiVersion: "v1"
+					kind:       "Namespace"
+					metadata: name: parameter.namespace
+				}]
+			},
+		]
+	}
+}
+```
+
+As we can see, this CUE file mainly contain a `ouput` block. The `output` block must be a KubeVela application that may contain components, policies, or workflow. 
+
+> Notice: This CUE file must have a `package main` header.
+
+In this example, we also define a namespace in `k8s-objects` component, and the name of it is set by parameters. If you want the created namespace is `my-namespace` you can execute the command like this:
 
-* [velaux](https://github.com/kubevela/catalog/blob/master/addons/velaux/template.yaml), define only the header of application in the template.
-* [OCM control plane](https://github.com/kubevela/catalog/blob/master/addons/ocm-hub-control-plane/template.yaml), the addon only be installed in control plane, so it just defines everything in the application template.
+```shell
+$ vela addon enable <addon-name> namespace=my-namespace
+```
 
-### `resources/` directory(Optional)
+After rendered, the resulting application is:
 
-Addon is actually an Application, but it provides more convenience for you to compose a complex application. You can use CUE or YAML to render resources into an application. This is what `resources` directory does for you.
+```yaml
+kind: Application
+metadata:
+  name: example
+  namespace: vela-system
+spec:
+  components:
+    - name: namespace
+      type: k8s-objects
+      properties:
+        objects:
+          - apiVersion: v1
+            kind: Namespace
+            metadata:
+              name: my-namespace
+```
 
-Files defined in `resources` folder eventually rendered as components in the application described in `template.yaml`.
+> Please notice: The name of Application in template file will be replaced by the addon name in `metadata.yaml`. The addon application will always have a unified name in the format of `addon-<addon_name>`.
 
-#### CUE format resource
+Although you can define the entire content of the application in the template file, but this may cause the template file to become very large, so you can choose to split a part of some content into the `resources` directory, latter section will introduce the all details.
```

Comment:
In case your template file is too large, you can split the entire content of the application into multiple parts and place them in files under the `resources` directly. The following section will discuss it in detail.